### PR TITLE
[Feature] create cloudnative table based on default warehouse

### PR DESCRIPTION
**Pull request by abc982627271**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -1810,7 +1811,20 @@ private List<CreateReplicaTask> buildCreateReplicaTasks(long dbId, OlapTable tab
             if (table.isCloudNativeTableOrMaterializedView()) {
                 long primaryComputeNodeId = -1;
                 try {
-                    primaryComputeNodeId = ((LakeTablet) tablet).getPrimaryComputeNodeId();
+                    if (Config.only_use_compute_node) {
+                        // it is ok for cloudnative table to build tablet in any warehouse
+                        Warehouse warehouse = GlobalStateMgr.getCurrentWarehouseMgr().
+                                getWarehouse(WarehouseManager.DEFAULT_WAREHOUSE_NAME);
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:1817]: how about the warehousemgr provide `getWarehouse()` interface and returns the default warehouse. The caller doesn't care about what the default ware house name is.

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java [hunk 2]
```diff
@@ -440,6 +455,44 @@ public long getPrimaryComputeNodeIdByShard(long shardId) throws UserException {
         throw new UserException("Failed to get primary backend. shard id: " + shardId);
     }
 
+    // the final func
+    public long getPrimaryComputeNodeIdByShard(long shardId, long workerGroupId) throws UserException {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:459]: should not repeat the whole implementation. 
```
public long getPrimaryComputeNodeIdByShard(shardId) {
    return getPrimaryComputeNodeIdByShard(shardId, DEFAULT_ID);
}
````

**Review by kevincai — DISMISSED** [review]

**Review by kevincai — DISMISSED** [review]
LGTM, need fix the code style reported by CI check.

**Review by wyb — DISMISSED** [review]

**Review by kevincai — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 22 / 38 (57.89%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|0|5|00.00%|[244, 245, 246, 249, 250]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|8|18|44.44%|[557, 559, 560, 562, 563, 564, 565, 566, 567, 568]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|3|4|75.00%|[1035]|
|:large_blue_circle:|com/starrocks/server/WarehouseManager.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/warehouse/Cluster.java|3|3|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java [hunk 3]
```diff
@@ -541,4 +551,21 @@ public boolean queryMetaGroupStable(long metaGroupId) {
         }
         return false; // return false if any error happens
     }
+
+    public long createWorkerGroup(String size) throws DdlException {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:555]: should be not required anymore.

File: fe/fe-core/src/main/java/com/starrocks/warehouse/Cluster.java [hunk 4]
```diff
@@ -37,17 +42,18 @@ public class Cluster implements Writable {
 
     public Cluster(long id) {
         this.id = id;
-    }
-
-    public Cluster(long id, long workerGroupId) {
-        this.id = id;
-        this.workerGroupId = workerGroupId;
+        // use default workerGroupId for community version
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/warehouse/Cluster.java:45]: remove this comment.

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java [hunk 5]
```diff
@@ -1194,6 +1190,11 @@ private void transferToLeader() {
             startAllNodeTypeDaemonThreads();
             insertOverwriteJobManager.cancelRunningJobs();
 
+            // must behind startLeaderOnlyDaemonThreads(), because creating workerGroup depends on serviceId
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:1193]: comment invalid now.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 22 / 28 (78.57%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|0|5|00.00%|[244, 245, 246, 249, 250]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|3|4|75.00%|[1035]|
|:large_blue_circle:|com/starrocks/server/WarehouseManager.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/warehouse/Cluster.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|8|8|100.00%|[]|

**Review by wyb — DISMISSED** [review]

**Review by sduzh — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 23 / 29 (79.31%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|0|5|00.00%|[244, 245, 246, 249, 250]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|3|4|75.00%|[1035]|
|:large_blue_circle:|com/starrocks/server/WarehouseManager.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/warehouse/Cluster.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|8|8|100.00%|[]|

**Review by kevincai — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**